service/discovery/azure/compute.go

```diff
@@ -121,7 +121,7 @@ func (d *azureComputeDiscovery) handleVirtualMachines(vm *compute.VirtualMachine
 
 	vmExtended, err := d.getExtendedVirtualMachine(vm)
 	if err != nil {
-		return nil, fmt.Errorf("could not get virtual machine: %w", err)
+		return nil, err
```

Comment:
Don't we have to pass on the error here instead of creating a new one?  If yes, then we need to check the whole code again.